assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -1232,5 +1238,371 @@ describe("DiversionPage", () => {
         screen.queryByRole("button", { name: "Change route or direction" })
       ).not.toBeInTheDocument()
     })
+
+    test("when in test group, is visible", async () => {
+      jest
+        .mocked(getTestGroups)
+        .mockReturnValue([TestGroups.DetourRouteSelection])
+      jest
+        .mocked(fetchRoutePatterns)
+        .mockResolvedValue(routePatternFactory.buildList(1, { id: "66_666" }))
+
+      render(
+        <DiversionPage originalRoute={{ routePattern: { id: "66_666" } }} />
+      )
+
+      expect(
+        await screen.findByRole("heading", { name: "Create Detour" })
+      ).toBeInTheDocument()
+      expect(
+        screen.queryByRole("button", { name: "Change route or direction" })
+      ).toBeInTheDocument()
+    })
+
+    test("when clicked, shows the route selection panel", async () => {
+      jest
+        .mocked(getTestGroups)
+        .mockReturnValue([TestGroups.DetourRouteSelection])
+
+      const routePatterns = routePatternFactory.buildList(1, { id: "66_666" })
+      const [routePattern] = routePatterns
+      jest.mocked(fetchRoutePatterns).mockResolvedValue(routePatterns)
+
+      render(<DiversionPage originalRoute={{ routePattern }} />)
+
+      await userEvent.click(
+        await screen.findByRole("button", { name: "Change route or direction" })
+      )
+
+      expect(
+        await screen.getByRole("heading", { name: "Choose route" })
+      ).toBeVisible()
+    })
+  })
+
+  describe("'Route Selection Panel'", () => {
+    beforeEach(() => {
```

Comment:
**Non-blocking**

In general, I don't love having a `beforeEach` in the middle of the test file, because it makes it a lot harder to tell what context each test is working with. For any of the tests within this `describe`, they've got an extra test group enabled, but it's hard to tell when looking at a given test which `describe` it's in.

A possible solution would be to enable this test group at the top for _all_ tests in this file, and then disable it only for the one test that asserts that the button isn't there (I think there's only one test that cares).

Non-blocking because this will be short-lived anyway, but food for thought.